### PR TITLE
Basic ripcord functionality

### PR DESCRIPTION
**Pull request by Boda805**
# **Pull Request Template**

###### *Select a category that relates to the content of your pull request (choose all that apply).*

- [ ] Bug Fix
- [ ] Content
- [x ] New Feature
- [ ] Documentation
- [ ] Code Refactoring

&nbsp;

## **Summary of Changes**

https://docs.google.com/document/d/1abPab88kII2WI1G7nbHlhGJs_hi2H-xH-bAjP6XyyhM/edit

Bare basic look so far

&nbsp;

## **Test Data or Screenshots**

image

&nbsp;

###### *By submitting this pull request, you are confirming the following to be true:*

- I have reviewed the [Contribution Guidelines](https://github.com/SetProtocol/index-ui/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `SetProtocol/index-ui`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by Boda805** [comment]
Are e2e tests out of date? I am getting some failures there but everything looks good on the site from what I can tell.

File: src/components/ProductPage/ProductRipcord.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  const formatLeverageMetric = (metricValue: number) =>
+    numeral(metricValue).format('0.00').toString().toUpperCase()
+
+  const realLeverage =
+    tokenData.components && tokenData.components.length > 0
+      ? Number(tokenData.components[0].percentOfSet || 0) / 100
+      : 2
+
+  const maxLeverage = Ethereum2xFlexibleLeverageIndex.symbol ? 2.3 : 2.2
```
Comment (Boda805) [inline src/components/ProductPage/ProductRipcord.tsx:32]: Could probably be improved upon, I don't love the hard-coding

**Comment by controtie** [comment]
> Are e2e tests out of date? I am getting some failures there but everything looks good on the site from what I can tell.

The e2e tests are really fragile atm. If you update the DOM the classnames all change and the tests break. Peeps is currently working on a fix for this. In the mean time can you address the tests? The PR looks mostly good to go.

File: src/components/ProductPage/ProductRipcord.tsx (hunk 1 above)
Comment (controtie) [reply src/components/ProductPage/ProductRipcord.tsx:32]: Can we make this a bit more explicit. Maybe set a variable at the top of the file or in a constants file.
`const BTC2xLeverageRatio = 2.2`
`const ETH2xLeverageRatio = 2.3`

**Comment by Boda805** [comment]
This should do the trick. Edited the max leverage amounts as well, according to:

https://docs.indexcoop.com/our-products/flexible-leverage-indices/fli-technical-documentation/fli-keeper-bot-integration

File: src/hooks/useRipcord.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      FlexibleLeverageStrategyABI as unknown as AbiItem,
+      flexibleLeverageStrategyExtension
+    )
+    try {
+      const ripcord: string = await ripcordContract.methods
+        .ripcord('UniswapV3ExchangeAdapter')
+        .call()
+      return ripcord
+    } catch (e) {
+      return '0'
```
Comment (controtie) [inline src/hooks/useRipcord.ts:23]: Seems a bit odd to return '0' here? What is this function's return value generally used for?

File: src/components/ProductPage/ProductRipcord.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      )
+    }
+    return (
+      <StyledNegativeChange>
+        {formatLeverageMetric(leverage)}x
+      </StyledNegativeChange>
+    )
+  }
+
+  return (
```
Comment (controtie) [inline src/components/ProductPage/ProductRipcord.tsx:51]: Can we make the styling in this section consistent with the `Market Data` section?
image

File: src/components/ProductPage/ProductRipcord.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          </StyledStatMetric>
+        </div>
+        <div>
+          <StyledStatTitle>Max Leverage</StyledStatTitle>
+          <StyledStatMetric>{maxLeverage.toString()}x</StyledStatMetric>
+        </div>
+        <div>
+          <SimpleButton
+            text='Pull Ripcord'
+            isDisabled={!isMetamaskConnected || realLeverage < maxLeverage}
```
Comment (controtie) [inline src/components/ProductPage/ProductRipcord.tsx:67]: Only disable this if metamask isn't connected. It looks like `realLeverage` is somewhat unreliably calculated. This is something of an emergency function, so we don't really want to block users from calling it (e.g. our users node provider might be behind)

**Review by controtie — CHANGES_REQUESTED** [review]
Implementation looks mostly good! Just some styling things can be cleaned up

**Pull request closed without merging**